common/src/stack/discovery/pylib/discovery.py

```diff
@@ -151,33 +151,20 @@ def _add_node(self, interface, mac_address, ip_address):
 		if network is not None:
 			# Add our new node
 			result = subprocess.run([
-				"/opt/stack/bin/stack",
-				"add",
-				"host",
-				self.hostname,
-				f"appliance={self._appliance_name}",
-				f"rack={self._rack}",
-				f"rank={self._rank}",
-				f"box={self._box}"
+				"/opt/stack/bin/stack", "add", "host", self.hostname,
+				f"rack={self._rack}", f"rank={self._rank}",
+				f"appliance={self._appliance_name}", f"box={self._box}"
 			], stdout=subprocess.PIPE, stderr=subprocess.PIPE, encoding="utf-8")
 
 			if result.returncode != 0:
 				self._logger.error("failed to add host %s:\n%s", self.hostname, result.stderr)
 				return
 
-			# Add the node's interface
+			# Add the node's interface (guess eth0 but will be updated pre-install)
 			result = subprocess.run([
-				"/opt/stack/bin/stack",
-				"add",
-				"host",
-				"interface",
-				self.hostname,
-				"interface=NULL",
-				"default=true",
-				f"mac={mac_address}",
-				f"name={self.hostname}",
-				f"ip={ip_address}",
-				f"network={network}"
+				"/opt/stack/bin/stack", "add", "host", "interface", self.hostname,
+				"interface=eth0", "default=true", f"mac={mac_address}",
+				f"name={self.hostname}", f"ip={ip_address}", f"network={network}"
```

Comment:
Consider adding a trailing comma, check throughout.

Reply:
I'm actually not a fan of the trailing comma, although I like that Python gives me the option.